back/app/services/surveys/average_generator.rb

```diff
@@ -0,0 +1,147 @@
+# frozen_string_literal: true
+
+module Surveys
+  class AverageGenerator
+    def initialize(phase, input_type: nil)
+      form = phase.custom_form || CustomForm.new(participation_context: phase)
+      @fields = IdeaCustomFieldsService.new(form).enabled_fields.select do |f|
+        f.input_type == input_type || f.supports_average? # Defaults to returning all fields that support averages
+      end
+      @inputs = phase.ideas.supports_survey.published
+      @phase = phase
+    end
+
+    # Averages for all fields by id
+    def field_averages(answers: all_answers)
+      @fields.to_h { |f| [f[:id], calculate_average(answers.pluck(f.key))] }
+    end
+
+    def field_averages_by_quarter
+      grouped_answers = all_answers.group_by { |a| a['quarter'] }
```

Comment:
Not quite sure how this works - but could there be any issue here when there is data for multiple years (E.g. Data for 2025-Q1 and 2026-Q1)?

Reply:
quarter in this instance is the full string of '2025-4'